components/RichTextEditor.js

```diff
@@ -330,58 +330,82 @@ export default class RichTextEditor extends React.Component {
     const { toolbarElement } = e.target;
     if (e.actionName === 'x-video-dialog-open') {
       const attachVideoDialog = toolbarElement.querySelector('[data-trix-dialog=video-url]');
-      const attachVideoDialogInput = toolbarElement.querySelector('.trix-input--dialog-video');
+      const attachVideoDialogInput = toolbarElement.querySelector('.trix-input--dialog-embed');
       if (attachVideoDialog.getAttribute('data-trix-active') === '') {
         attachVideoDialog.removeAttribute('data-trix-active');
       } else {
         attachVideoDialog.setAttribute('data-trix-active', '');
         attachVideoDialogInput.removeAttribute('disabled');
       }
-    } else if (e.actionName === 'x-video-embed') {
-      const videoLink = toolbarElement.querySelector('.trix-input--dialog-video').value;
-      if (videoLink) {
-        this.embedVideoIFrame(videoLink);
+    } else if (e.actionName === 'x-add-embed') {
+      const embedLink = toolbarElement.querySelector('.trix-input--dialog-embed').value?.trim();
+      if (embedLink) {
+        this.embedIframe(embedLink);
       }
     }
   };
 
-  constructVideoEmbedURL = videoLink => {
-    const { id, service } = this.parseVideoLink(videoLink);
+  constructVideoEmbedURL = (service, id) => {
     if (service === 'youtube') {
       return `https://www.youtube-nocookie.com/embed/${id}`;
     } /* else if (service === 'vimeo') {
       return `https://player.vimeo.com/video/${id}`;
-    }*/ else {
+    } */ else if (
+      service === 'anchorFm'
+    ) {
+      return `https://anchor.fm/${id}`;
+    } else {
       return null;
     }
   };
 
-  parseVideoLink = videoLink => {
+  parseServiceLink = videoLink => {
     const regexps = {
       youtube: new RegExp(
-        '(?:https?://)?(?:www.)?youtu(?:.be/|be.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)',
-        'ig',
+        '(?:https?://)?(?:www\\.)?youtu(?:\\.be/|be\\.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)',
```

Comment:
Edited YouTube and Vimeo regexps to escape dots. before that, `https://vimeoxcom/xxxx` was considered as a valid Vimeo link.